client/file-browser/get-files.ts

```diff
@@ -13,34 +13,36 @@ function splitExtension(filename: string): [base: string, ext: string] {
 }
 
 export default async function readFolder(folderPath: string): Promise<FileBrowserEntry[]> {
-  try {
-    const entries = (await ipcRenderer.invoke('fsReadDir', folderPath, { withFileTypes: true }))!
-    return await Promise.all(
-      entries.map(async entry => {
-        const isFolder = entry.isDirectory
-        const [name, extension] = isFolder ? [entry.name, ''] : splitExtension(entry.name)
-        const stats = await ipcRenderer.invoke('fsStat', folderPath + '\\' + entry.name)
-        const path = folderPath + '\\' + entry.name
-
-        return isFolder
-          ? {
-              type: FileBrowserEntryType.Folder,
-              name,
-              path,
-            }
-          : {
-              type: FileBrowserEntryType.File,
-              name,
-              path,
-              extension: extension.toLowerCase(),
-              date: stats?.mtime ?? new Date(0),
-            }
-      }),
-    )
-  } catch (err) {
-    if ((err as any).code === 'ENOENT') {
-      throw new Error('Filepath ' + folderPath + " doesn't exist")
-    }
-    throw err
+  const readDirResult = (await ipcRenderer.invoke('fsReadDir', folderPath, {
+    withFileTypes: true,
+  }))!
+  if (readDirResult.isError && readDirResult.errorCode !== FsErrorCode.ENOENT) {
+    throw new Error('Unexpected FsError: ' + readDirResult.errorCode)
   }
+  const entries: FsDirent[] = readDirResult.isError ? [] : readDirResult.payload!
```

Comment:
This is the biggest divergence in behavior, instead of returning an error on a non-existent directory. For it's existing usage in `file-browser`, I think that's correct but I could see it not being a safe assumption in the future.